mlrun/common/schemas/artifact.py

```diff
@@ -51,10 +53,43 @@ class ArtifactIdentifier(pydantic.BaseModel):
     key: typing.Optional[str]
     iter: typing.Optional[int]
     uid: typing.Optional[str]
+    tree: typing.Optional[str]
```

Comment:
2 things:
1. does it make sense to use tree instead of producer id?
2. is the comment below still relevant after the refactor?